### ISSUE TITLE
"Delete selected row" button is disabled when navigating the rows of a table using the keyboard

### ISSUE THREAD

**Issue by Ivan-Perez** [HeidiSQL/HeidiSQL#624]
## Steps to reproduce this issue

1. Go to the data tab of any table.
2. Click on a row. At this moment, the delete button for that row and its keyboard shortcut (Ctrl+Del) are enabled.
3. Move to another row using the keyboard (i.e. press down arrow).
4. The delete button is no longer enabled, nor is its shortcut. Clicking on that row or on any other reenables it.

A gif is worth a thousand words. Look at the delete button, it isn't always enabled:

![bug](https://i.imgur.com/oEYag8B.gif)

## Current behavior
Already described.

## Expected behavior
The button should always be enabled.

## Possible solution
.

## Environment

I don't remember if this behaviour didn't exist in a previous version, but as far as I can remember, this issue is happening to me since long ago (I would bet one year at least, if not since ever), but haven't decided to open this bug report until now :)

* **HeidiSQL version:**  10.1.0.5545
* **Database system and version:** MariaDB 5.5.60
* **Operating system:** Windows 10

**Comment by ansgarbecker** [comment]
Also, selecting rows by mouse via dragging a rectangle disables the delete button.

**Comment by ansgarbecker** [comment]
I had to trigger control/button validation also in the grid's OnChange event, which is fired quite often, so cross fingers this does not harm performance.

**Comment by ansgarbecker** [comment]
By the way, I am curious: how did you create that screen-animation from above?

**Comment by Ivan-Perez** [comment]
> 
> 
> I had to trigger control/button validation also in the grid's OnChange event, which is fired quite often, so cross fingers this does not harm performance.

Thanks, it works perfectly! I don't see any performance loss at the moment.

> 
> 
> By the way, I am curious: how did you create that screen-animation from above?

I used the first tool I found on Google. In this case, it was [ScreenToGif](https://www.screentogif.com/).

By the way, I think [Gyazo](https://gyazo.com/captures) is the most popular tool to make GIFs/MP4 taken from the screen, but I didn't remember its name when I opened this issue so I tried another program.
